Starcraft2Client.py

```diff
@@ -568,169 +561,116 @@ async def on_step(self, iteration: int):
-                        await self.ctx.send_msgs(
-                            [{"cmd": 'LocationChecks',
-                              "locations": [SC2WOL_LOC_ID_OFFSET + 100 * self.mission_id + 5]}])
-                        self.fifth_bonus = True
-
-                    if not self.sixth_bonus and game_state & (1 << 7):
-                        print("6th Bonus Collected")
-                        await self.ctx.send_msgs(
-                            [{"cmd": 'LocationChecks',
-                              "locations": [SC2WOL_LOC_ID_OFFSET + 100 * self.mission_id + 6]}])
-                        self.sixth_bonus = True
-
-                    if not self.seventh_bonus and game_state & (1 << 8):
-                        print("6th Bonus Collected")
-                        await self.ctx.send_msgs(
-                            [{"cmd": 'LocationChecks',
-                              "locations": [SC2WOL_LOC_ID_OFFSET + 100 * self.mission_id + 7]}])
-                        self.seventh_bonus = True
-
-                    if not self.eight_bonus and game_state & (1 << 9):
-                        print("6th Bonus Collected")
-                        await self.ctx.send_msgs(
-                            [{"cmd": 'LocationChecks',
-                              "locations": [SC2WOL_LOC_ID_OFFSET + 100 * self.mission_id + 8]}])
-                        self.eight_bonus = True
+                    for x, completed in enumerate(self.boni):
+                        if not completed and game_state & (1 << (x + 2)):
+                            await self.ctx.send_msgs(
+                                [{"cmd": 'LocationChecks',
+                                  "locations": [SC2WOL_LOC_ID_OFFSET + victory_modulo * self.mission_id + x + 1]}])
+                            self.boni[x] = True
 
                 else:
                     await self.chat_send("LostConnection - Lost connection to game.")
 
 
-def calc_objectives_completed(mission, missions_info, locations_done, unfinished_locations, ctx):
+def calc_objectives_completed(ctx: SC2Context, mission: str) -> int:
     objectives_complete = 0
+    objectives_exist = 0
 
-    if missions_info[mission].extra_locations > 0:
-        for i in range(missions_info[mission].extra_locations):
-            if (missions_info[mission].id * 100 + SC2WOL_LOC_ID_OFFSET + i) in locations_done:
+    for i in range(max_bonus+1):
+        loc_id: int = ctx.mission_req_table[mission].id * victory_modulo + SC2WOL_LOC_ID_OFFSET + i
+        if loc_id in ctx.server_locations:
+            objectives_exist += 1
+            if loc_id in ctx.checked_locations:
                 objectives_complete += 1
-            else:
-                unfinished_locations[mission].append(ctx.location_names[
-                    missions_info[mission].id * 100 + SC2WOL_LOC_ID_OFFSET + i])
-
-        return objectives_complete
 
-    else:
+    if objectives_exist == 0:
         return -1
```

Comment:
i believe this is unused, does not really hurt though

Reply:
All-in is marked like this

Reply:
```python
        else:  # infer that this is the final mission as it has no objectives
```
for that code path, calc_objectives_completed is never called

Reply:
Correct, as there are none.